patches/0007-Update-default-go.env.patch

```diff
@@ -0,0 +1,25 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: qmuntal <qmuntaldiaz@microsoft.com>
+Date: Thu, 8 Jun 2023 14:17:13 +0200
+Subject: [PATCH] Update default go.env
+
+---
+ go.env | 8 +++++---
+ 1 file changed, 5 insertions(+), 3 deletions(-)
+
+diff --git a/go.env b/go.env
+index 9bab8ffd73dd71..4aca2ff2c63dbc 100644
+--- a/go.env
++++ b/go.env
+@@ -7,6 +7,8 @@
+ GOPROXY=https://proxy.golang.org,direct
+ GOSUMDB=sum.golang.org
+ 
+-# Automatically download newer toolchains as directed by go.mod files.
+-# See https://go.dev/doc/toolchain for details.
+-GOTOOLCHAIN=auto
++# Use the locally installed Go toolchain, never downloading a different one.
++# Upstream uses `GOTOOLCHAIN=auto` instead, but `auto` can download and switch
++# to a non-MSFT Go toolchain, and we want to avoid that.
++# See https://go.dev/doc/toolchain for details.
```

Comment:
Note for reviewers: go.dev/doc/toolchain is still not available, draft here https://go-review.googlesource.com/c/website/+/500775.